src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/Specifiable.cs

```diff
@@ -0,0 +1,45 @@
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Common
+{
+    /// <summary>
+    /// Represents a value which may or may not be specified.
+    /// </summary>
+    /// <remarks>
+    /// This is independent of whether the value is null or not;
+    /// a value may be specified as null.
+    /// </remarks>
+    /// <typeparam name="T">The type of the value to be specified.</typeparam>
+    public struct Specifiable<T>
+    {
+        /// <summary>
+        /// Whether the value is specified.
+        /// </summary>
+        public bool IsSpecified;
+
+        private T _value;
+
+        /// <summary>
+        /// Initializes a new instance of a specifiable value interpreted as
+        /// being specified.
+        /// </summary>
+        /// <remarks>
+        /// To create an unspecified instance, use the default (parameterless)
```

Comment:
Don't you need to explicitly declare the parameterless constructor as well? 
I thought the default constructor *went away* if you defined your own.

Reply:
I will confess that I have only recently got the hang of working with `struct` - got my head truly confuzzled years ago and worked exclusively with `class` for a *long* time.

Reply:
Structs will always have it.

Reply:
Huh, somehow the intermediate comments were missing when I wrote that... 😁